### PR TITLE
Fix coverage with multiple expr groups producing the same val group

### PR DESCRIPTION
**Pull request by naoyam**
Previously, in `computeCoveredGroups`, when `covered_ids` has already a mapping, the additional covered groups are simply ignored. This could cause the loop promotion analysis to fail to find any promotion. See the added test for a concrete example. That repro is rather synthetic, but the same issue also happens with some of the reshape shmoo tests.

### CODE REVIEW

**Comment by naoyam** [comment]
!build

File: csrc/id_model/loop_promotion.cpp [hunk 1]
```diff
@@ -693,6 +693,13 @@ std::unordered_map<ValGroup, ValGroups> computeCoveredGroups(
       // Don't overwrite initialized cases due to rfactor markings.
       if (covered_ids.find(output_group) == covered_ids.end()) {
         covered_ids[output_group] = covered;
+      } else {
+        // An exact group may have multiple exact expr groups and may
+        // have different coverage groups depending on the expr
+        // groups. For example, this can happen with reshape or
+        // resize. See test LoopPromotionCoverage for a concrete
+        // example.
+        covered_ids[output_group].pushBack(covered);
```
Comment (zasdfgbnm) [inline csrc/id_model/loop_promotion.cpp:702]: Can this be simplified to just a single:
```C++
for (const ValGroup& output_group : graph.outputGroups(exact_expr)) {
  covered_ids[output_group].pushBack(covered);
}
```

File: csrc/id_model/loop_promotion.cpp [hunk 2]
```diff
@@ -693,6 +693,13 @@ std::unordered_map<ValGroup, ValGroups> computeCoveredGroups(
       // Don't overwrite initialized cases due to rfactor markings.
       if (covered_ids.find(output_group) == covered_ids.end()) {
         covered_ids[output_group] = covered;
+      } else {
```
Comment (zasdfgbnm) [inline csrc/id_model/loop_promotion.cpp:696]: Is `view_rfactor_ids` still used?

Comment (naoyam) [reply csrc/id_model/loop_promotion.cpp:696]: Good point. Removed.

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**